pkg/validate/validate.go

```diff
@@ -0,0 +1,49 @@
+package validate
+
+import (
+	"fmt"
+
+	"github.com/mt-sre/addon-metadata-operator/api/v1alpha1"
+)
+
+type Validator struct {
+	Description string
+	Runner      ValidateFunc
+}
+
+type ValidateFunc func(mb *MetaBundle) error
+
+type MetaBundle struct {
+	AddonMeta *v1alpha1.AddonMetadataSpec
+	// TODO: add field for corresponding bundle
+}
+
+// TODO: This will return a MetaBundle with corresponding bundle
+func NewMetaBundle(addonMeta *v1alpha1.AddonMetadataSpec) *MetaBundle {
+	return &MetaBundle{
+		AddonMeta: addonMeta,
+	}
+}
+
+func (mb *MetaBundle) Validate(runMeta bool) []error {
+	errs := []error{}
+
+	if runMeta {
+		validators := GetAllMetaValidators()
+
+		printMetaHeading()
```

Comment:
[nit] Should we use logger instead of print to stdout? Thinking about when this code will run inside of a `ValidationWebhook`. Or we detect this somehow, and JSON log if running inside a `ValidationWebhook`, to better integrate /w fluentd/elasticsearch etc. This is a problem for future us probably.

Reply:
This is a really good point. I agree with having a standardized log format to be propagated in this project.  We can make use of logrus for example, it provides JSON formatted logging out of the box. Or we can use any existing logging library (logger/logr iirc) which is already bootstrapped by kubebuilder but would have confirm once about the extent of their formatting capabilities.